webui/react/src/stores/users.tsx

```diff
@@ -45,27 +54,27 @@ export const UsersProvider: React.FC<PropsWithChildren> = ({ children }) => {
   );
 };
 
-export const useFetchUsers = (canceler: AbortController): (() => Promise<void>) => {
+export const useFetchUsers = (canceler: AbortController): ((cfg?: FetchUsersConfig) => void) => {
   const context = useContext(UsersContext);
 
   if (context === null) {
     throw new Error('Attempted to use useFetchUsers outside of Users Context');
   }
 
-  const { updateUsers } = context;
+  const { users, updateUsers } = context;
 
   return useCallback(
-    async (cfg?: FetchUsersConfig): Promise<void> => {
+    async (cfg?: FetchUsersConfig) => {
       try {
         const config = cfg ?? {};
         const response = await getUsers(config, { signal: canceler.signal });
 
-        updateUsers(() => Loaded(response.users));
+        updateUsers(users.set(encodeParams(config), response));
```

Comment:
`users.set(encodeParams(config)` should be assigned to other variable first for readability